Sources/MongoSwift/MongoClient.swift

```diff
@@ -304,6 +304,107 @@ public class MongoClient {
+      * - Parameters:
+      *   - pipeline: An array of aggregation pipeline stages to apply to the events returned by the change stream.
+      *   - options: An optional `ChangeStreamOptions` to use on the initial `ChangeStream` stage.
+      *   - session: An optional `ClientSession` to use with this change stream.
+      * - Returns: a `ChangeStream` on all collections in all databases in a cluster.
+      * - Throws:
+      *   - `ServerError.commandError` if an error occurs on the server while creating the change stream.
+      *   - `ServerError.commandError` if the pipeline passed is invalid.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      * - SeeAlso:
+      *   - https://docs.mongodb.com/manual/changeStreams/
+      *   - https://docs.mongodb.com/manual/meta/aggregation-quick-reference/
+      *   - https://docs.mongodb.com/manual/reference/system-collections/
+      * - Note: Supported in MongoDB version 4.0+ only.
+      */
+     public func watch(_  pipeline: [Document] = [],
+                       options: ChangeStreamOptions?  =  nil,
+                       session: ClientSession? = nil) throws ->
+                       ChangeStream<ChangeStreamDocument<Document>> {
+        return try self.watch(pipeline, options: options, session: session, withFullDocumentType: Document.self)
+     }
+
+     /**
+      * Starts a `ChangeStream` on a `MongoClient`. Allows the client to observe all changes in a cluster - excluding
+      * system collections i.g. "config", "local", and "admin" databases. Associates the specified `Codable` type `T`
+      * with the `fullDocument` field in the `ChangeStreamDocument` emitted by the returned `ChangeStream`.
+      * - Parameters:
+      *   - pipeline: An array of aggregation pipeline stages to apply to the events returned by the change stream.
+      *   - options: An optional `ChangeStreamOptions` to use on the initial `ChangeStream` stage.
+      *   - session: An optional `ClientSession` to use with this change stream.
+      *   - withFullDocumentType: The type that the change events emitted from the change stream will be decoded to.
+      * - Returns: A `ChangeStream` on all collections in all databases in a cluster.
+      * - Throws:
+      *   - `ServerError.commandError` if an error occurs on the server while creating the change stream.
+      *   - `ServerError.commandError` if the pipeline passed is invalid.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      *   - `DecodingError` if an error occurs while decoding user-defined `withFullDocumentType` `Codable` type.
```

Comment:
would we see the decoding errors now? it seems to me like they wouldn't occur until the user calls `next()` on the change stream.